resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java

```diff
@@ -300,28 +300,31 @@ private String cnameResolveFromCache(String name) {
         return cnameResolveFromCacheLoop(cnameCache, first, second);
     }
 
-    private String cnameResolveFromCacheLoop(DnsCnameCache cnameCache, String first, String mapping) {
-        // Detect loops using a HashSet. We use this as last resort implementation to reduce allocations in the most
-        // common cases.
-        Set<String> cnames = new HashSet<String>(4);
-        cnames.add(first);
-        cnames.add(mapping);
+    static String cnameResolveFromCacheLoop(DnsCnameCache cnameCache, String first, String mapping) {
+        // Detect loops by advance only every other iteration.
+        // See https://en.wikipedia.org/wiki/Cycle_detection#Floyd's_Tortoise_and_Hare
+        String otherMapping = first;
```

Comment:
Just out of curiosity ,why you need field  otherMapping  to instead of first  which you never use .

Reply:
@prgitpr yep we can simplify.